lxd/auth/drivers/openfga.go

```diff
@@ -256,11 +256,14 @@ func (e *embeddedOpenFGA) CheckPermission(ctx context.Context, entityURL *api.UR
 
 	// If not allowed, decide if the user can view the resource.
 	if !resp.GetAllowed() {
+		err := auth.ValidateEntitlement(entityType, auth.EntitlementCanView)
```

Comment:
Should we be checking for a specific type/class of error here? What if ValidateEntitlement later returns an error which doesn't mean "no error, but user cant view", but means "something went wrong internally"?